### DISCUSSION TITLE
Fit TFN in parts of data

### DISCUSSION THREAD

**Discussion by bgeorgios** [pastas/pastas#309] [Q&A] [answered]
Hi,

First of all, thank you very much for the nice work you've put on this package! 

I am trying to use Pastas to quantify the contribution of rain and snowmelt to the GWL time series. Say I have continuous daily data for 10 years. Is there a way that I fit a single model to the winter data i.e. October-March, solely?

For example, use October 2010-March 2011, October 2011-March 2012 etc., without the use of the intermediate summer periods.

Thanks,
Georgios

---
Ph.D. student at the University of Alabama, Tuscaloosa, AL.
[gboumis.com](https://www.gboumis.com)

**Comment by raoulcollenteur** [comment, marked as answer]
Hi @bgeorgios! Thanks for checking out the Pastas package. 

There is a solution for this, by using `weights` argument in `ml.solve()`. [Find the docs here](https://pastas.readthedocs.io/en/latest/api/generated/generated/pastas.model.Model.solve.html). So what you can do is provide a Pandas Series with binary values (e.g., 0 for the summer and 1 for the winter). Then all observations where the weight is zero are not taken into account during calibration. I hope this works, the `weights` feature has not been fully tested but it should work, otherwise let me know :)

Cheers,
Raoul

**Reply by bgeorgios** [reply]
Thanks I will try it out!

**Reply by raoulcollenteur** [reply]
Hi Georgios, curious to hear if that actually helped for your problem. Did using the weights work? Cheers, Raoul

**Reply by cnicol-gwlogic** [reply]
For the record, this is what worked for me and yep Raoul it works as expected: 
```
weights = ml.observations(tmin, tmax, ml.settings["freq_obs"])
weights.loc[:]  = 1.0
``` 
...then adjust that as you like, and feed it to model.solve(weights=weights). The weights need to be specific to the solve tmin/max/freq_obs; I found this in ```pastas.model.residuals()``` code (this bit:  ```oseries_calib = self.observations(tmin, tmax, freq_obs)```).

**Reply by raoulcollenteur** [reply]
Thanks for the feedback @cnicol-gwlogic ! Happy to here it works!

Cheers,
Raoul
